src/actions/order-actions.js

```diff
@@ -198,37 +187,61 @@ function prepareShippingMethod (shippingMethod) {
   }
 }
 
-function discountSummaries (cart) {
+function prepareDiscountSummaries (cart, lineItems) {
   return cart.discount_summaries.map(discountSummary => {
+    const lineItemDiscounts = lineItems.map(lineItem => {
+      return lineItem.attributes.line_item_discounts.filter(lineItemDiscount => lineItemDiscount.attributes.promotion_id === discountSummary.promotion_id)
+    }).flat()
+
     return {
       id: discountSummary.id,
       type: 'discount_summaries',
       attributes: {
         promotion_id: discountSummary.promotion_id,
         name: discountSummary.name,
-        total: discountSummary.total
+        amount_ex_tax: lineItemDiscounts.reduce((sum, discount) => sum + discount.attributes.amount_ex_tax, 0),
+        amount_inc_tax: lineItemDiscounts.reduce((sum, discount) => sum + discount.attributes.amount_inc_tax, 0)
       }
     }
   })
 }
 
-// To be tested and used when changes to v2 endpoint are deployed
-//
-// function discountSummaries (cart, lineItems) {
-//   return cart.discount_summaries.map(discountSummary => {
-//     const lineItemDiscounts = lineItems.map(lineItem => {
-//       return lineItem.attributes.line_item_discounts.find(lineItemDiscount => lineItemDiscount.promotion_id === discountSummary.promotion_id)
-//     }).filter(discount => discount)
-
-//     return {
-//       id: discountSummary.id,
-//       type: 'discount_summaries',
-//       attributes: {
-//         promotion_id: discountSummary.promotion_id,
-//         name: discountSummary.name,
-//         amount_ex_tax: lineItemDiscounts.reduce((discountA, discountB) => discountA.amount_ex_tax + discountB.amount_ex_tax, 0),
-//         amount_inc_tax: lineItemDiscounts.reduce((discountA, discountB) => discountA.amount_inc_tax + discountB.amount_inc_tax, 0)
-//       }
-//     }
-//   })
-// }
+function roundDecimals (order) {
```

Comment:
Rounding is done as the very last step so that we don't lose pennies when doing maths.